network/dag/publisher.go

```diff
@@ -124,11 +124,17 @@ func (s *replayingDAGPublisher) publish(ctx context.Context) {
 }
 
 func (s *replayingDAGPublisher) publishTransaction(ctx context.Context, transaction Transaction) bool {
+	// We need to skip transactions with a to addr header as it should be handled by the v2 protocol
```

Comment:
Shouldn't this be the place where a future event should be generated?
Is this the place where the `to` header decrypt should be performed? (If for me => check if payload is present)

Just some comments for future committers.

Reply:
> Shouldn't this be the place where a future event should be generated? Is this the place where the `to` header decrypt should be performed? (If for me => check if payload is present)
> 
> Just some comments for future committers.

I think the `to` header should be decrypted by a listener (Nats/observer/whatever) in protocol v2.

Reply:
I guess it *should* be decrypted here right? If that succeeds it can be pushed to an event listener in the v2 protocol.

Reply:
so that will be a broader discussion ;)